### PR TITLE
refactor: change to class, add unit tests

### PR DESCRIPTION
**Pull request by AnguIar**

### CODE REVIEW

File: src/cleanup_script.js [hunk 1]
```diff
@@ -1,150 +1,154 @@
...
+  async deleteFiles(pathArray) {
+    const promiseDeleteArray = pathArray.map((path) => promises.unlink(path));
+    return Promise.all(promiseDeleteArray);
+  }
 
-async function s3MainDeleteLoop(notCleaned) {
-  for (const discrete of notCleaned) {
-    const { id, version } = discrete;
-    await deleteS3WithBatch(`${id}/${version}`);
+  async deleteChunks(urisArray) {
```
Comment (asafMasa) [inline src/cleanup_script.js:46]: rename to deleteFsChunks

File: src/cleanup_script.js [hunk 2]
```diff
@@ -1,150 +1,154 @@
...
+      ContinuationToken = prepareItems.ContinuationToken;
+    }
+  }
 
-async function deleteMapProxyLayer(discreteLayers) {
-  const mapproxyUrl = config.get('mapproxy_api').url;
-  const mapProxyLayersToDelete = [];
-  for (const discrete of discreteLayers) {
-    mapProxyLayersToDelete.push(axios.delete(`${mapproxyUrl}/layer/${discrete.id}-${discrete.version}`));
+  async parseItemsFromS3(Prefix, ContinuationToken) {
```
Comment (asafMasa) [inline src/cleanup_script.js:84]: it seams as actual delete not parse, am i wrong? rename is needed?

File: src/cleanup_script.js [hunk 3]
```diff
@@ -1,150 +1,154 @@
...
-  const BATCH_SIZE = config.get('batchSize').discreteLayers;
-
-  for (let i = 0; i < notCleanedAndFailed.length; i += BATCH_SIZE) {
-    const currentBatch = notCleanedAndFailed.slice(i, i + BATCH_SIZE);
-    await deleteOriginalFiles(currentBatch);
-    await s3MainDeleteLoop(currentBatch);
-    await deleteMapProxyLayer(currentBatch);
-    await markAsCompleted(currentBatch);
+  async deleteMapProxyLayer(discreteLayers) {
+    const mapproxyUrl = config.get('mapproxy_api').url;
```
Comment (asafMasa) [inline src/cleanup_script.js:114]: add info log of the layer that is about to be deleted

Comment (asafMasa) [reply src/cleanup_script.js:114]: do we need to check if the layer exists before trrying to delete it?

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good job!!
liked the refactor 💣

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const CleanupScript = require('./cleanup_script');
+const { getLoggerInstance } = require('./logger');
+const logger = getLoggerInstance();
+
+new CleanupScript()
```
[the remark this answers is not in the document]
Comment (AnguIar) [reply src/index.js:5]: This doesn't work. If you want to use `try-catch` you'll need `await` - but you _cant_ `await` in global scope.

File: src/cleanup_script.js (hunk 2 above)
Comment (AnguIar) [reply src/cleanup_script.js:84]: It is one step before delete. Deletion from `s3` takes place in `deleteFromS3`

File: src/cleanup_script.js (hunk 3 above)
Comment (AnguIar) [reply src/cleanup_script.js:114]: There is a log

File: src/cleanup_script.js [hunk 5]
```diff
@@ -119,13 +114,19 @@ class CleanupScript {
     this.logger.info(
       `Deleting layers [${discreteLayers.map((discrete) => `${discrete.id}-${discrete.version}`)}] from mapproxy in path [${mapproxyUrl}]`
     );
-    await Promise.all(mapProxyLayersToDelete);
+    try {
+      await Promise.all(mapProxyLayersToDelete);
+    } catch (err) {
+      this.logger.info(
```
Comment (asafMasa) [inline src/cleanup_script.js:120]: seperate Not_Found to other exceptions

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**